src/CMakeLists.txt

```diff
@@ -1 +1 @@
-idf_component_register(SRCS "main.cpp" "asdf.cpp") #"abstract_task.cpp")
+idf_component_register(SRCS "main.cpp" "../interfaces/abstract_task.cpp")
```

Comment:
We should actually be a able to register the abstrac task as a separate idf component. You can try to add an empty `component.mk` file as well as a `CMakeLists.txt` file with the following content:
```
idf_component_register(
        SRCS "abstract_task.cpp"
        INCLUDE_DIRS ".")
```
into the `interfaces` folder.

Reply:
done. (by you ;)